### PR TITLE
'getPopulation and complementPopulation methods were created'

### PR DESCRIPTION
**Pull request by opapina**

### CODE REVIEW

File: src/main/java/com/solvd/schooltimetable/persistence/GenericAlgo.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+import com.solvd.schooltimetable.service.impl.TeacherServiceImpl;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import java.util.stream.Collectors;
+
+public class GenericAlgo {
```
Comment (amuravski) [inline src/main/java/com/solvd/schooltimetable/persistence/GenericAlgo.java:20]: replace letter 'r' in class name with 't'

File: src/main/java/com/solvd/schooltimetable/persistence/GenericAlgo.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+    private static final Logger LOGGER = LogManager.getLogger(GenericAlgo.class);
+    private static GeneticAlgoConfig geneticAlgoConfig = new GeneticAlgoConfig("geneticAlgoConfig.properties");
```
Comment (amuravski) [inline src/main/java/com/solvd/schooltimetable/persistence/GenericAlgo.java:23]: remove initialization

File: src/main/java/com/solvd/schooltimetable/persistence/GenericAlgo.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        for (int i = 0; i < geneticAlgoConfig.getPopulationSize(); i++) {
+            schoolTimetables.add(getRandomSchoolTimetable());
+        }
+        return schoolTimetables;
+    }
+
+    public static List<SchoolTimetable> complementPopulation(List<SchoolTimetable> schoolTimetables) {
+        int currentPopulationSize = schoolTimetables.size();
+        int complementPopulationSize = geneticAlgoConfig.getPopulationSize() - currentPopulationSize;
+        for (int i = 0; i < complementPopulationSize; i++) {
```
Comment (amuravski) [inline src/main/java/com/solvd/schooltimetable/persistence/GenericAlgo.java:77]: replace with IntStream.range

File: src/main/java/com/solvd/schooltimetable/persistence/GenericAlgo.java (hunk 2 above)
Comment (opapina) [reply src/main/java/com/solvd/schooltimetable/persistence/GenericAlgo.java]: fixed for LOGGER

**Review by amuravski — APPROVED** [review]

**Pull request merged**